internal/configs/config.go

```diff
@@ -1039,6 +1048,38 @@ func (c *Config) getProviderConfigTransformForTest(evalCtx *hcl.EvalContext) tes
 	}
 }
 
+// evaluateProviderConfig evaluates code for the mock provider. In the current implementation,
+// it only evaluates the for_each expression.
```

Comment:
tiny tiny nitpick: you mention that "in the current implementation" it does something, is there ever going to be a future implementation? I don't think we need this kind of comment here.